features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/KeyManagers/ClaimMapping.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState } from 'react';
+import TextField from '@material-ui/core/TextField';
+import { makeStyles } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import IconButton from '@material-ui/core/IconButton';
+import DeleteIcon from '@material-ui/icons/Delete';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+import ClearIcon from '@material-ui/icons/Clear';
+import Alert from 'AppComponents/Shared/Alert';
+
+const useStyles = makeStyles((theme) => ({
+    mandatoryStar: {
+        color: theme.palette.error.main,
+        marginLeft: theme.spacing(0.1),
+    },
+}));
+
+/**
+ * Trottling Policies dropdown selector used in minimized API Create form
+ * @export
+ * @param {*} props
+ * @returns {React.Component}
+ */
+export default function ClaimMappings(props) {
+    const classes = useStyles();
+    const [newRemoteClaim, setRemoteClaim] = useState('');
+    const [newLocalClaim, setLocalClaim] = useState('');
+    const { claimMappings, setClaimMapping } = props;
+    const [validationError, setValidationError] = useState([]);
+
+    const onChange = (e) => {
+        if (e.target.id === 'remoteClaim') {
+            setRemoteClaim(e.target.value);
+        } else if (e.target.id === 'localClaim') {
+            setLocalClaim(e.target.value);
+        }
+    };
+    const validate = (fieldName, value) => {
+        let error = '';
+        if (value === null || value === '') {
+            error = 'Calim is Empty';
+        } else {
+            error = '';
+        }
+        setValidationError({ fieldName: error });
+        return error;
+    };
+
+    const clearValues = () => {
+        setLocalClaim(null);
+        setRemoteClaim(null);
+    };
+    const handleAddToList = () => {
+        const remoteClaimError = validate('remoteClaim', newRemoteClaim);
+        const localClaimError = validate('localClaim', newLocalClaim);
+        if (localClaimError !== '' || remoteClaimError !== '') {
+            Alert.error(remoteClaimError);
+            clearValues();
+            return false;
+        } else {
+            let exist = false;
+            claimMappings.map(({ remoteClaim, localClaim }) => {
```

Comment:
I think here we are looking for at least one `exist`?
If so we can use `for of` loop to iterate the `claimMappings` and break on first `exist`